subprojects/frameworklauncher/yarn/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -1361,8 +1387,21 @@ public void onError(Throwable e) {
   }
 
   public void onShutdownRequest() {
-    stopForInternalTransientNormalError(
-        "onShutdownRequest called into AM from RM, maybe this Attempt does not exist in RM.");
+    Boolean isApplicationStopped = false;
+    try {
+      isApplicationStopped = statusManager.getFrameworkExecutionType() == ExecutionType.STOP;
+    } catch (Throwable e) {
+      LOGGER.logWarning(e, "Failed to check framework execution type, AM will not unregister");
```

Comment:
You catch all things, it may be a breaking change if RMClient calls onShutdownRequest

Reply:
No, in the case of any errors, old behaviour is saved.